xonsh/prompt/vc_branch.py

```diff
@@ -125,9 +125,9 @@ def current_branch(pad=NotImplemented):
                       "and will be removed in the future")
     branch = None
     cmds = builtins.__xonsh_commands_cache__
-    if cmds.lazy_locate_binary('git'):
+    if cmds.lazy_locate_binary('git') or cmds.locate_binary('git'):
         branch = get_git_branch()
-    if cmds.lazy_locate_binary('hg') and not branch:
+    if cmds.lazy_locate_binary('hg') or cmds.locate_binary('hg') and not branch:
```

Comment:
you're right but i am not sure about using threads for updating the cache, maybe there is another thread which messes with the cache at the same time. caching will happen anyway at some point so we may want to do it on a higher level than this module.